src/Execution/ExecutionStrategy.php

```diff
@@ -189,7 +189,7 @@ private function doesFragmentConditionMatch(
 
         $conditionalType = typeFromAST($this->context->getSchema(), $typeConditionNode);
 
-        if ($conditionalType === $type) {
+        if (gettype($conditionalType) === gettype($type)) {
```

Comment:
I'm not a fan of this solution. Simply comparing the instances should work.

Reply:
The issue is It didn't work :<